doc/Makefile.am

```diff
@@ -148,6 +148,75 @@ CLEAN_LOCAL_TARGETS += clean-gtkdoc-header-local
 clean-gtkdoc-header-local:
 	-rm -rf xml/ Doxyfile-gi Doxyfile-gi.stamp geany-gtkdoc.h geany-sciwrappers-gtkdoc.h
 
+if HAVE_INTROSPECTION
+
+INTROSPECTION_GIRS  = Geany-1.0-pre.gir GeanyScintilla-1.0-pre.gir
```

Comment:
For the sciwrappers part, because g-ir-scanner can't be told to put method prefixed with `sci` into the `ScintillaObject` namespace, so have nice API (i.e. `sci.start_undo_action()`) the code here renames the functions to a `scinitilla_object` prefix and strip them back to have the proper name.  Yep, it's a hack.

Not sure about the `GeanyScintillaScintillaObject` thing.